DuckDuckGo/TabViewControllerLongPressBookmarkExtension.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  TabViewControllerLongPressBookmarkExtension.swift
+//  DuckDuckGo
+//
+//  Created by BG on 2/15/19.
+//  Copyright © 2019 DuckDuckGo. All rights reserved.
+//
+
+import Foundation
+import Core
+
+extension TabViewController {
+    func promptSaveBookmarkAction() {
+        
+        if let link = link, !isError {
+            let bookmarksManager = BookmarksManager()
+            guard !bookmarksManager.contains(url: link.url) else {
+                Logger.log(text: "promptSave already bookmarked")
+                self.view.showBottomToast(UserText.webBookmarkAlreadySaved)
+                return
+            }
+            
+            let saveCompletion: (Link) -> Void = { [weak self] updatedBookmark in
+                bookmarksManager.save(bookmark: updatedBookmark)
+                self?.view.showBottomToast(UserText.webSaveBookmarkDone)
+            }
+            let alert = EditBookmarkAlert.buildAlert (
+                title: UserText.alertSaveBookmark,
+                bookmark: link,
+                saveCompletion: saveCompletion)
+            self.present(alert, animated: true, completion: nil)
+        } else {
+            Logger.log(text: "Invalid bookmark link found on bookmark long press")
```

Comment:
I don't think we need to log anything in this method - these are quite common scenarios.

Reply:
i think this error conditions would be useful if we ever get into such situations, unless there is already some error handling when such scenario happens.
As i know, these logging messages are only enabled in debug build which meant for developer..